tests/unit/hook_packages/terraform/hooks/prepare/test_resource_linking.py

```diff
@@ -131,3 +132,144 @@ def test_resource_full_address_root_module(self):
         module = TFModule(None, None, {}, {}, {}, {})
         resource = TFResource("resource_address", "type", module, {})
         self.assertEqual(resource.full_address, "resource_address")
+
+    def test_resolve_module_variable_constant_value(self):
+        constant_value = ConstantValue(value="layer.arn")
+        module = TFModule(
+            variables={"layer": constant_value},
+            resources=[],
+            child_modules={},
+            outputs={},
+            full_address="",
+            parent_module=None,
+        )
+        results = _resolve_module_variable(module, "layer")
+        expected_result = "layer.arn"
+        for result in results:
+            self.assertEqual(result.value, expected_result)
+
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._get_configuration_address")
+    @patch("samcli.hook_packages.terraform.hooks.prepare.resource_linking._clean_references_list")
+    def test_resolve_module_variable_nested_variables(self, mock_clean_references, mock_get_configuration_address):
+        references = ["var.layer_name"]
+        mock_clean_references.return_value = references
+        mock_get_configuration_address.return_value = "layer_name"
+        references = References(value=references)
+        constant_value = ConstantValue(value="layer.arn")
+        parent_module = TFModule(
+            variables={"layer_name": constant_value},
+            resources=[],
+            child_modules={},
+            outputs={},
+            full_address="",
+            parent_module=None,
+        )
+        child_module = TFModule(
+            variables={"layer": references},
+            resources=[],
+            child_modules={},
+            outputs={},
+            full_address="",
+            parent_module=parent_module,
+        )
+        results = _resolve_module_variable(child_module, "layer")
+        expected_result = "layer.arn"
+        for result in results:
+            self.assertEqual(result.value, expected_result)
```

Comment:
can we replace it with validating the list as we expect .. this loop can pass if there are multiple items returned in the loop but all of them are the same value.